### PR TITLE
Hyperlink support

### PR DESCRIPTION
**Pull request by meain**
Fixes https://github.com/Peltoche/lsd/issues/192 . There is still some more code cleanup left, but thought I would push the what I had so far.

Thanks to @KSXGitHub for the original implementation in https://github.com/Peltoche/lsd/pull/354.

---
#### TODO

- [x] Use `cargo fmt`
- [x] Add necessary tests
- [x] Add changelog entry
- [x] Update default config/theme in README (if applicable)
- [x] Update man page at lsd/doc/lsd.md (if applicable)

### CODE REVIEW

**Comment by meain** [comment]
I have some cleanup left in this, but have been a bit busy the past few weeks. Hope to compete this soon.

**Comment by meain** [comment]
This should be good to go now, there is some possible flags rework we could do, but I think I'll take that up later once I have a concrete idea. https://github.com/Peltoche/lsd/blob/ca5d9b3e160b0d9e6e1f1d2717e9df7cc51814e4/src/core.rs#L56

File: doc/lsd.md [hunk 1]
```diff
@@ -107,6 +107,9 @@ lsd is a ls command with a lot of pretty colours and some other stuff to enrich
 `--group-directories-first`
 : Groups the directories at the top before the files. Same as `--group-dirs=first`
 
+`--hyperlink <hyperlink>...`
+: Attach hyperlink to filenames [default: auto]  [possible values: always, auto, never]
```
Comment (zwpaper) [inline doc/lsd.md:111]: the default value should be `never`?

Comment (meain) [reply doc/lsd.md]: Good catch.

**Pull request merged**